api/storage_service.cc

```diff
@@ -1403,16 +1411,28 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
         } else {
             throw httpd::bad_param_exception(fmt::format("Unknown argument for 'quarantine_mode' parameter: {}", quarantine_mode_str));
         }
-        return f.then([&ctx, keyspace, column_families, opts] {
-            return ctx.db.invoke_on_all([=] (replica::database& db) {
-                return do_for_each(column_families, [=, &db](sstring cfname) {
+
+        const auto& reduce_compaction_counters = [] (const compaction_manager::compaction_counters_opt& lhs, const compaction_manager::compaction_counters_opt& rhs) {
+            sstables::compaction_counters counters{};
+            counters += lhs.value();
+            counters += rhs.value();
+            return counters;
+        };
```

Comment:
You can just add `operator+()` to `compaction_counters`, then use `std::plus<compaction_counters>` in the `map_reduce` below.

Reply:
Oh, it is an optional. This would work OOTB if we used plain values instead of optional.